### ISSUE TITLE
Support py3

### ISSUE THREAD

**Issue by joth76** [deepmind/multidim-image-augmentation#3]
Split out from issue #2 

Currently only py2.7 is supported, this issue tracks adding py3 support.
 
see "undefined symbol: PyCObject_Type" below

```
ERROR: /home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/external/org_tensorflow/ten
sorflow/BUILD:592:1: Executing genrule @org_tensorflow//tensorflow:tensorflow_python_api_gen failed (Exit 1)
Traceback (most recent call last):
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_au
gmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorfl
ow/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_au
gmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorfl
ow/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "/home/erfannoury/anaconda3/lib/python3.6/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/home/erfannoury/anaconda3/lib/python3.6/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: /home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/_pywrap_tensorflow_internal.so: undefined symbol: PyCObject_Type

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/tools/api/generator/create_python_api.py", line 27, in <module>
    from tensorflow.python.tools.api.generator import doc_srcs
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/pywrap_tensorflow.py", line 74, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "/home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "/home/erfannoury/anaconda3/lib/python3.6/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/home/erfannoury/anaconda3/lib/python3.6/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: /home/erfannoury/.cache/bazel/_bazel_erfannoury/2f5f4c4cfb6ca03d34e4841831474c3a/execroot/multidim_image_augmentation/bazel-out/host/bin/external/org_tensorflow/tensorflow/create_tensorflow.python_api.runfiles/org_tensorflow/tensorflow/python/_pywrap_tensorflow_internal.so: undefined symbol: PyCObject_Type
```

**Comment by joeyearsley** [comment]
I have this working on PY3, only thing which failed for me was `deformation_utils_test.py` as highlighted in my PR.

This seems like it might be more of a Bazel issue, I had numerous with Bazel and the default setup (can't depend on core:framework, building without cuda support, having to reinstall TF as it rebuilt the lib with a new string which wasn't in the default lib_tensorflow.so)

I've now made it work, for pretty much any TF you have installed through pip, in the current way Tensorflow recommends through:
https://github.com/tensorflow/custom-op

Will try and post a PR as it might help some people.

**Comment by joeyearsley** [comment]
https://github.com/joeyearsley/multidim-image-augmentation

On my fork's master I've altered this code base:
 * To work with the pre-installed TF pip package 
* Removed the cc_tests; quickest way to build since I removed a lot of external dependencies, use the py_tests.
* Can't generate a python file automatically so you will need to write the documentation and interface manually - since I removed the dependency of `tf_gen_op_wrapper_py`

**Comment by joth76** [comment]
Fixed in f3dde76
